common/azure-common-auth/src/test/java/com/azure/common/auth/credentials/http/MockHttpClient.java

```diff
@@ -17,13 +18,20 @@
  * This HttpClient attempts to mimic the behavior of http://httpbin.org without ever making a network call.
  */
 public class MockHttpClient implements HttpClient {
-    private static final HttpResponse mockResponse = new MockHttpResponse(200);
     private final List<HttpRequest> requests;
 
+    /**
+     * Creates a new MockHttpClient that mimics http://httbin.org.
+     */
     public MockHttpClient() {
         requests = new ArrayList<>();
     }
 
+    /**
+     * Gets the requests sent by MockHttpClient.
+     *
+     * @return The requests sent by this HTTP client.
+     */
```

Comment:
We should remove the requirement that public unit test methods require JavaDoc.

Reply:
I added a rule here that ignores Javadoc for classes that start with `Mock*`. Is that sufficient? There are existing rules that exclude `*Test`.
https://github.com/Azure/azure-sdk-for-java/pull/3412/files#diff-8ab8d3eeaecf610d7802a6417b5bcdf5R16

Reply:
I would just apply it to anything under `/src/test/java`

Reply:
Does this work? https://github.com/Azure/azure-sdk-for-java/pull/3412/files?file-filters%5B%5D=.xml#diff-8ab8d3eeaecf610d7802a6417b5bcdf5R17

I removed the ones that had overlapping logic.